### ISSUE TITLE
3.0.0-alpha9 nmake test failures windows x64

### ISSUE THREAD

**Issue by supernoodled** [openssl/openssl#13474]
build is as of commit 5811387bac39cdb6d009dc0139b56e6896259cbd.

vers:
Windows 10 Pro version 20H2, build 19042.630 x64
Strawberry Perl 5.32.0.1 x64
NASM 2.15.05 x64
Visual Studio 2017

used:
x64 Native Tools Command Prompt for VS 2017 in admin mode
then perl configured with VC-WIN64A.

all tests ok other than this one:

```20-test_dgst.t ..................... 2/?
C:\Strawberry\perl\bin\perl.exe ..\..\util\wrap.pl ..\..\apps\openssl.exe dgst -sha256 -hmac 123456 ..\..\test\data.txt ..\..\test\data.txt => 0
    not ok 1 - HMAC: Check HMAC value is as expected (HMAC-SHA256(..\..\test\data.txt)= 53bf88b80462d52356ba3b017b86014edbfb38763319276e60e404614e238b94) vs ((?^:HMAC-SHA256\([^\)]*data.txt\)= 6f12484129c4a761747f13d8234a1ff0e074adb34e9e9bf3a155c391b97b9a7c))
# ------------------------------------------------------------------------------
    #   Failed test 'HMAC: Check HMAC value is as expected (HMAC-SHA256(..\..\test\data.txt)= 53bf88b80462d52356ba3b017b86014edbfb38763319276e60e404614e238b94) vs ((?^:HMAC-SHA256\([^\)]*data.txt\)= 6f12484129c4a761747f13d8234a1ff0e074adb34e9e9bf3a155c391b97b9a7c))'
    #   at test\recipes\20-test_dgst.t line 115.
    not ok 2 - HMAC: Check second HMAC value is consistent with the first (HMAC-SHA256(..\..\test\data.txt)= 53bf88b80462d52356ba3b017b86014edbfb38763319276e60e404614e238b94) vs ((?^:HMAC-SHA256\([^\)]*data.txt\)= 6f12484129c4a761747f13d8234a1ff0e074adb34e9e9bf3a155c391b97b9a7c))
# ------------------------------------------------------------------------------
    #   Failed test 'HMAC: Check second HMAC value is consistent with the first (HMAC-SHA256(..\..\test\data.txt)= 53bf88b80462d52356ba3b017b86014edbfb38763319276e60e404614e238b94) vs ((?^:HMAC-SHA256\([^\)]*data.txt\)= 6f12484129c4a761747f13d8234a1ff0e074adb34e9e9bf3a155c391b97b9a7c))'
    #   at test\recipes\20-test_dgst.t line 116.
    # Looks like you failed 2 tests of 2.
not ok 6 - HMAC generation with `dgst` CLI
# ------------------------------------------------------------------------------
#   Failed test 'HMAC generation with `dgst` CLI'
#   at C:\Users\noodled\source\repos\openssl\util\perl/OpenSSL/Test.pm line 1335.
20-test_dgst.t ..................... Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/6 subtests
        (less 2 skipped subtests: 3 okay)



Test Summary Report
-------------------
20-test_dgst.t                   (Wstat: 256 Tests: 6 Failed: 1)
  Failed test:  6
  Non-zero exit status: 1
Files=224, Tests=3430, 1427 wallclock secs (16.83 usr +  2.12 sys = 18.95 CPU)
Result: FAIL
NMAKE : fatal error U1077: 'C:\Strawberry\perl\bin\perl.exe' : return code '0x1'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\bin\HostX64\x64\nmake.exe"' : return code '0x2'
Stop.


perl configdata.pm --dump

Command line (with current working directory = .):

    C:\Strawberry\perl\bin\perl.exe Configure VC-WIN64A

Perl information:

    C:\Strawberry\perl\bin\perl.exe
    5.32.0 for MSWin32-x64-multi-thread

Enabled features:

    acvp_tests
    aria
    asm
    async
    autoalginit
    autoerrinit
    autoload-config
    bf
    blake2
    camellia
    capieng
    cast
    chacha
    cmac
    cmp
    cms
    comp
    ct
    deprecated
    des
    dgram
    dh
    dsa
    dso
    dtls
    dynamic-engine
    ec
    ec2m
    ecdh
    ecdsa
    engine
    err
    filenames
    fips
    fips-securitychecks
    gost
    idea
    legacy
    makedepend
    md4
    mdc2
    module
    multiblock
    nextprotoneg
    pinshared
    ocb
    ocsp
    padlockeng
    pic
    poly1305
    posix-io
    psk
    rc2
    rc4
    rdrand
    rfc3779
    rmd160
    scrypt
    secure-memory
    seed
    shared
    siphash
    siv
    sm2
    sm3
    sm4
    sock
    srp
    srtp
    sse2
    ssl
    static-engine
    stdio
    tests
    threads
    tls
    ts
    ui-console
    uplink
    whirlpool
    tls1
    tls1-method
    tls1_1
    tls1_1-method
    tls1_2
    tls1_2-method
    tls1_3
    dtls1
    dtls1-method
    dtls1_2
    dtls1_2-method

Disabled features:

    afalgeng            [not-linux] OPENSSL_NO_AFALGENG
    asan                [default]   OPENSSL_NO_ASAN
    buildtest-c++       [default]
    crypto-mdebug       [default]   OPENSSL_NO_CRYPTO_MDEBUG
    devcryptoeng        [default]   OPENSSL_NO_DEVCRYPTOENG
    ec_nistp_64_gcc_128 [default]   OPENSSL_NO_EC_NISTP_64_GCC_128
    egd                 [default]   OPENSSL_NO_EGD
    external-tests      [default]   OPENSSL_NO_EXTERNAL_TESTS
    fuzz-libfuzzer      [default]   OPENSSL_NO_FUZZ_LIBFUZZER
    fuzz-afl            [default]   OPENSSL_NO_FUZZ_AFL
    ktls                [default]   OPENSSL_NO_KTLS
    md2                 [default]   OPENSSL_NO_MD2 (skip crypto\md2)
    msan                [default]   OPENSSL_NO_MSAN
    rc5                 [default]   OPENSSL_NO_RC5 (skip crypto\rc5)
    sctp                [default]   OPENSSL_NO_SCTP
    ssl-trace           [default]   OPENSSL_NO_SSL_TRACE
    trace               [default]   OPENSSL_NO_TRACE
    ubsan               [default]   OPENSSL_NO_UBSAN
    unit-test           [default]   OPENSSL_NO_UNIT_TEST
    weak-ssl-ciphers    [default]   OPENSSL_NO_WEAK_SSL_CIPHERS
    zlib                [default]
    zlib-dynamic        [default]
    ssl3                [default]   OPENSSL_NO_SSL3
    ssl3-method         [default]   OPENSSL_NO_SSL3_METHOD

Config target attributes:

    AR => "lib",
    ARFLAGS => "/nologo",
    AS => "nasm",
    ASFLAGS => "-g",
    CC => "cl",
    CFLAGS => "/W3 /wd4090 /nologo /O2",
    CPP => "\$(CC) /EP /C",
    HASHBANGPERL => "/usr/bin/env perl",
    LD => "link",
    LDFLAGS => "/nologo /debug",
    MT => "mt",
    MTFLAGS => "-nologo",
    RANLIB => "CODE(0xf143a0)",
    RC => "rc",
    aroutflag => "/out:",
    asflags => "-Ox -f win64 -DNEAR",
    asm_arch => "x86_64",
    asoutflag => "-o ",
    bin_cflags => "/Zi /Fdapp.pdb",
    bin_lflags => "/subsystem:console /opt:ref",
    bn_ops => "SIXTY_FOUR_BIT",
    build_file => "makefile",
    build_scheme => [ "unified", "windows", "VC-common" ],
    cflags => "/Gs0 /GF /Gy /MD",
    coutflag => "/Fo",
    cppflags => "",
    defines => [ "OPENSSL_BUILDING_OPENSSL", "OPENSSL_SYS_WIN32", "WIN32_LEAN_AND_MEAN", "UNICODE", "_UNICODE", "_CRT_SECURE_NO_DEPRECATE", "_WINSOCK_DEPRECATED_NO_WARNINGS" ],
    disable => [  ],
    dso_cflags => "/Zi /Fddso.pdb",
    dso_scheme => "win32",
    enable => [  ],
    ex_libs => "ws2_32.lib gdi32.lib advapi32.lib crypt32.lib user32.lib",
    includes => [  ],
    ldoutflag => "/out:",
    lflags => "",
    lib_cflags => "/Zi /Fdossl_static.pdb",
    lib_cppflags => "",
    lib_defines => [ "L_ENDIAN" ],
    module_cflags => "",
    module_cxxflags => undef,
    module_ldflags => "/dll",
    mtinflag => "-manifest ",
    mtoutflag => "-outputresource:",
    multilib => "-x64",
    perl_platform => "Windows::MSVC",
    perlasm_scheme => "auto",
    rcoutflag => "/fo",
    shared_cflag => "",
    shared_defflag => "",
    shared_defines => [  ],
    shared_ldflag => "/dll",
    shared_rcflag => "",
    shared_target => "win-shared",
    sys_id => "WIN64A",
    thread_defines => [  ],
    thread_scheme => "winthreads",
    unistd => "<unistd.h>",
    uplink_arch => "x86_64",

Recorded environment:

    AR =
    ARFLAGS =
    AS =
    ASFLAGS =
    BUILDFILE =
    CC =
    CFLAGS =
    CPP =
    CPPDEFINES =
    CPPFLAGS =
    CPPINCLUDES =
    CROSS_COMPILE =
    CXX =
    CXXFLAGS =
    HASHBANGPERL =
    LD =
    LDFLAGS =
    LDLIBS =
    MT =
    MTFLAGS =
    OPENSSL_LOCAL_CONFIG_DIR =
    PERL =
    RANLIB =
    RC =
    RCFLAGS =
    RM =
    WINDRES =
    __CNF_CFLAGS =
    __CNF_CPPDEFINES =
    __CNF_CPPFLAGS =
    __CNF_CPPINCLUDES =
    __CNF_CXXFLAGS =
    __CNF_LDFLAGS =
    __CNF_LDLIBS =

Makevars:

    AR              = lib
    ARFLAGS         = /nologo
    AS              = nasm
    ASFLAGS         = -g
    CC              = cl
    CFLAGS          = /W3 /wd4090 /nologo /O2
    CPP             = $(CC) /EP /C
    CPPDEFINES      =
    CPPFLAGS        =
    CPPINCLUDES     =
    CXXFLAGS        =
    HASHBANGPERL    = /usr/bin/env perl
    LD              = link
    LDFLAGS         = /nologo /debug
    LDLIBS          =
    MT              = mt
    MTFLAGS         = -nologo
    PERL            = C:\Strawberry\perl\bin\perl.exe
    RANLIB          = ranlib
    RC              = rc
    RCFLAGS         =

NOTE: These variables only represent the configuration view.  The build file
template may have processed these variables further, please have a look at the
build file for more exact data:
    makefile

build file:

    makefile

build file templates:

    Configurations\common0.tmpl
    Configurations\windows-makefile.tmpl
    Configurations\common.tmpl```

**Comment by levitte** [comment]
Heh...  text files.  Different line ending on Windows (CRLF) than on Unix (LF).  Of course the result is different.

Fix coming up.

**Comment by levitte** [comment]
Fix in #13477, please try it out
